cmd/genslateyaml/gen_slate_yaml.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v2"
+
+	"k8s.io/kubernetes/pkg/kubectl/cmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+// gen_slate_yaml creates a yaml representation of the kubectl help commands.  This is to be consumed
+// by tools to generate documentation.
+
+var outputFile = flag.String("output-file", "", "Destination for kubectl yaml representation.")
+
+func main() {
+	flag.Parse()
+
+	if len(*outputFile) < 1 {
+		fmt.Printf("Must specify --output-file.\n")
+		os.Exit(2)
```

Comment:
any special reason to use `2` here? From [http://www.tldp.org/LDP/abs/html/exitcodes.html](http://www.tldp.org/LDP/abs/html/exitcodes.html), seems exit code 2 has special meaning.